zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/store/ElasticsearchBatchRequestTest.java

```diff
@@ -0,0 +1,349 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.exporter.store;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import co.elastic.clients.elasticsearch.ElasticsearchClient;
+import co.elastic.clients.elasticsearch._types.ElasticsearchException;
+import co.elastic.clients.elasticsearch._types.ErrorCause;
+import co.elastic.clients.elasticsearch._types.Refresh;
+import co.elastic.clients.elasticsearch._types.Script;
+import co.elastic.clients.elasticsearch.core.BulkRequest;
+import co.elastic.clients.elasticsearch.core.BulkRequest.Builder;
+import co.elastic.clients.elasticsearch.core.BulkResponse;
+import co.elastic.clients.elasticsearch.core.bulk.BulkOperation;
+import co.elastic.clients.elasticsearch.core.bulk.BulkResponseItem;
+import co.elastic.clients.elasticsearch.core.bulk.IndexOperation;
+import io.camunda.exporter.entities.TestExporterEntity;
+import io.camunda.exporter.exceptions.PersistenceException;
+import io.camunda.exporter.utils.ElasticsearchScriptBuilder;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import org.assertj.core.api.ThrowableAssert.ThrowingCallable;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.mockito.ArgumentCaptor;
+
+class ElasticsearchBatchRequestTest {
+
+  private static final String ID = "id";
+  private static final String INDEX = "index";
+  private ElasticsearchBatchRequest batchRequest;
+  private ElasticsearchClient elasticsearchClient;
+  private Builder requestBuilder;
+  private ElasticsearchScriptBuilder scriptBuilder;
+
+  @BeforeEach
+  void setUp() {
+    elasticsearchClient = mock(ElasticsearchClient.class);
+    requestBuilder = mock(Builder.class);
+    scriptBuilder = mock(ElasticsearchScriptBuilder.class);
+    batchRequest =
+        new ElasticsearchBatchRequest(elasticsearchClient, requestBuilder, scriptBuilder);
+  }
+
+  @Test
+  void shouldAddABulkOperationWithSpecifiedIndexAndEntity() throws PersistenceException {
```

Comment:
I don't have a better suggestion on how to test this. But I was thinking more on the following line:

Instead of inspecting what operation is build using the requestBuilder, we can verify what request is being sent to ElasticSearch. This is relying on less implementation details. Something similar to the following. 

```
 @Test
  void shouldAddABulkOperationWithSpecifiedIndexAndEntity() throws PersistenceException, IOException {
    // Given
    final var requestBuilder = new Builder();
    final var batchRequest =
        new ElasticsearchBatchRequest(elasticsearchClient, requestBuilder, scriptBuilder);
    final TestExporterEntity entity = new TestExporterEntity().setId(ID);

    // When
    batchRequest.add(INDEX, entity);
    batchRequest.execute();
    
    // Then
    final ArgumentCaptor<BulkRequest> captor = ArgumentCaptor.forClass(BulkRequest.class);
    verify(elasticsearchClient).bulk(captor.capture());
    
    // verify that an index operation is added
    final var bulkOperation = captor.capture().operations().getFirst();
    assertThat(bulkOperation.isIndex()).isTrue();
    final IndexOperation<TestExporterEntity> index = bulkOperation.index();
    assertThat(index.index()).isEqualTo(INDEX);
    assertThat(index.id()).isEqualTo(ID);
    assertThat(index.document()).isEqualTo(entity);
  }
```